block/manager.go

```diff
@@ -607,6 +600,34 @@ func (m *Manager) SyncLoop(ctx context.Context, cancel context.CancelFunc) {
 			m.sendNonBlockingSignalToHeaderStoreCh()
 			m.sendNonBlockingSignalToRetrieveCh()
 
+			// check if the dataHash is dataHashForEmptyTxs
+			// no need to wait for syncing Data, instead prepare now and set
+			// so that trySyncNextBlock can progress
+			if bytes.Equal(header.DataHash, dataHashForEmptyTxs) {
```

Comment:
[non-blocking]
I think it is worthwhile making this a small helper function for this comparison so that it has a unit test where the bytes are hardcoded in the test. This would protect against the bytes changing and impacting this internal function logic.